tests/resources/test_generative_design_execution.py

```diff
@@ -0,0 +1,133 @@
+    return collection.build(generative_design_execution_dict)
+
+
+def test_basic_methods(generative_design_execution, collection):
+    assert "GenerativeDesignExecution" in str(generative_design_execution)
+
+    with pytest.raises(NotImplementedError):
+        collection.register(generative_design_execution)
+
+    with pytest.raises(NotImplementedError):
+        collection.update(generative_design_execution)
+
+
+def test_build_new_execution(collection, generative_design_execution_dict):
+    # Given
+    generative_design_execution_id = uuid.uuid4()
+    build_data = generative_design_execution_dict.copy()
+    build_data["id"] = str(generative_design_execution_id)
+
+    # When
+    execution: GenerativeDesignExecution = collection.build(build_data)
+
+    # Then
+    assert execution.uid == generative_design_execution_id
+    assert execution.project_id == collection.project_id
+    assert execution._session == collection.session
+    assert execution.in_progress() and not execution.succeeded() and not execution.failed()
+
+
+def test_trigger_execution(collection: GenerativeDesignExecutionCollection, generative_design_execution_dict, session):
+    # Given
+    session.set_response(generative_design_execution_dict)
+    design_execuption_input = GenerativeDesignInput(
+        seeds = ["CC(O)=O"],
+        fingerprint_type = FingerprintType.ECFP4,
+        min_fingerprint_similarity = 0.5,
+        mutation_per_seed = 2
+    )
+
+    # When
+    actual_execution = collection.trigger(design_execuption_input)
+
+    # Then
+    assert str(actual_execution.uid) == generative_design_execution_dict["id"]
+    expected_path = '/projects/{}/generative-design/executions'.format(
+        collection.project_id,
+    )
+    assert session.last_call == FakeCall(
+        method='POST',
+        path=expected_path,
+        json={
+            'seeds': design_execuption_input.seeds,
+            'fingerprint_type': design_execuption_input.fingerprint_type.value,
+            'min_fingerprint_similarity': design_execuption_input.min_fingerprint_similarity,
+            'mutation_per_seed': design_execuption_input.mutation_per_seed
+        }
+    )
+
+
+def test_generative_design_execution_results(generative_design_execution: GenerativeDesignExecution, session, example_generation_results):
+    # Given
+    session.set_response(example_generation_results)
+
+    # When
+    list(generative_design_execution.results(per_page=4))
+
+    # Then
+    expected_path = '/projects/{}/generative-design/executions/{}/results'.format(
+        generative_design_execution.project_id,
+        generative_design_execution.uid,
+    )
+    assert session.last_call == FakeCall(method='GET', path=expected_path, params={"per_page": 4})
+
+
+def test_generative_design_execution_result(generative_design_execution: GenerativeDesignExecution, session, example_generation_results):
+    # Given
+    session.set_response(example_generation_results)
```

Comment:
It strikes me as odd that `example_generation_results` is being used as the expected response for testing both the `results` and `result` function. The former returns a paginated list of results whereas the latter returns a single generative design result. So don't they have different json responses?

Reply:
Ah. I am grabbing the `id` from the first result in the `example_generation_results`. Thats why it is used in both locations.

Reply:
But why is it the response in both cases? That would imply that both `/results` and `/result/{id}` return json of the same shape. Do they?

Reply:
I see. Let me look at the backend API to double check what it returns.